worktree.go

```diff
@@ -897,6 +899,31 @@ func (w *Worktree) Submodule(name string) (*Submodule, error) {
 	return nil, ErrSubmoduleNotFound
 }
 
+// resolveModuleURL resolves relative URLs based on the originURL.
+// The moduleURL is returned verbatim when originURL is empty or moduleURL is an absolute URL.
+func resolveModuleURL(originURL, moduleURL string) (string, error) {
+	if originURL == "" {
+		return moduleURL, nil
+	}
+	if !strings.HasPrefix(moduleURL, "../") && !strings.HasPrefix(moduleURL, "./") {
+		return moduleURL, nil
+	}
+	if !giturl.MatchesScheme(originURL) && giturl.MatchesScpLike(originURL) {
+		user, host, portStr, path := giturl.FindScpLikeComponents(originURL)
+		path = filepath.Join(path, moduleURL)
+		if portStr != "" {
+			portStr += ":"
+		}
+		return fmt.Sprintf("%s@%s:%s%s", user, host, portStr, path), nil
```

Comment:
For SCP-like paths, this might not be enough, but it should be easier to debug this without having to run in Windows:

```suggestion
		user, host, portStr, p := giturl.FindScpLikeComponents(originURL)
		p = path.Join(path, moduleURL)
		if portStr != "" {
			portStr += ":"
		}
		return fmt.Sprintf("%s@%s:%s%s", user, host, portStr, p), nil
```

Reply:
Fixed via [2fbaf8f](https://github.com/go-git/go-git/pull/1581/commits/2fbaf8fabb8d57bff1884ef6958dacec8207e4d2) .